### PR TITLE
adding an option to change headerKey see issue #155

### PR DESCRIPTION
**Pull request by benjaminlees**
@nelsonic please review

### CODE REVIEW

File: lib/extract.js [hunk 1]
```diff
@@ -9,17 +9,18 @@ module.exports = function (request, options) {
   // The key holding token value in url or cookie defaults to token
   var urlKey = typeof options.urlKey === 'string' ? options.urlKey : 'token';
   var cookieKey = typeof options.cookieKey === 'string' ? options.cookieKey : 'token';
+  var headerKey = typeof options.headerKey === 'string' ? options.cookieKey : 'authorization';
```
Comment (nelsonic) [inline lib/extract.js]: @benjaminlees should it be:
```js
var headerKey = typeof options.headerKey === 'string' ? options.headerKey : 'authorization';
```
?

**Comment by nelsonic** [comment]
@benjaminlees we need to add a set of tests before we can merge/publish this.

File: lib/extract.js (hunk 1 above)
Comment (benjaminlees) [inline lib/extract.js]: just setting a default, is there something wrong ?

**Comment by benjaminlees** [comment]
ok i will add some tests

**Comment by benjaminlees** [comment]
@nelsonic i have added a test but i can not commit due to code coverage issues. which is odd as i was able to commit previously without a test

**Comment by benjaminlees** [comment]
i ran  npm run report and it is flagging a bunch of lines that i have not changed

**Comment by nelsonic** [comment]
hmm... ah. its because your test is requiring a server and you may not be fully testing that server...

**Comment by benjaminlees** [comment]
oh ok so what should i do ?

**Comment by benjaminlees** [comment]
wait i fixed it

**Comment by benjaminlees** [comment]
@nelsonic please review

**Comment by nelsonic** [comment]
@benjaminlees looks good.

**Pull request merged**